platform/src/Playground.js

```diff
@@ -358,7 +375,7 @@ function createPanelForDefinitionId(panel){
             // No activity defined buttons
             newPanel.addButtons( Button.createButtons( panelDefinition.buttons, panel.id));
 
-        } else if (panel.buttons != null && panelDefinition.buttons != null) {
+        } else if (panel.buttons != null && panelDefinition.buttons == null) {
```

Comment:
This change seems unrelated?

Reply:
A CompositePanel definition may not define buttons

```
 panelDefs:
    - id: composite-panel
      panelclass: CompositePanel
      icon: diagram
```

And these are defined in the activity:

```
 panels:
      - id: panel-yamtl
        name: Transformation(YAMTL)
        ref: yamtl
        file: CD2DB.groovy
      - id: smm-panel-composite
        name: Source Metamodel
        ref: composite-panel
        childrenPanels:
          - id: panel-smm
            name: EMFatic
            ref: emfatic
            file: CD.emf
          - id: panel-smm-diagram
            name: Diagram
            ref: plantuml
        buttons:
          - id: show-editor-button
            icon: editor
            internal: toggle
            targetPanel: panel-smm
            hint: Toggle editor
          - id: show-diagram-button
            icon: diagram
            internal: toggle
            targetPanel: panel-smm-diagram
            hint: Toggle diagram
```

The case where the panelDefinition had no buttons was not considered:

```
        if (panel.buttons == null && panelDefinition.buttons != null){

        } else if (panel.buttons != null && panelDefinition.buttons == null) {
```

Having said that, I have not tested the case `panel.buttons != null && panelDefinition.buttons != null`, where I the panel should override the buttons I guess.